gk/main.c

```diff
@@ -431,8 +485,20 @@ add_ggu_policy(struct ggu_policy *policy, struct gk_instance *instance)
 		fe->u.granted.budget_byte =
 			fe->u.granted.tx_rate_kb_cycle * 1024;
 
-		/* TODO Fill up the grantor id field. */
+		/* Grantor ID comes from LPM lookup. */
+		gt_id = get_grantor_id(ltbl, &fe->flow);
+		if (gt_id < 0) {
+			RTE_LOG(ERR, GATEKEEPER,
+				"gk: initialize flow entry error in %s\n",
+				__func__);
+			fe->u.granted.grantor_id = 0;
```

Comment:
First of all, we have two cases here. One is when the flow entry already exists; for this case, the grantor ID may be already known and we can save the lookup. The code must save this lookup because the lookup to find a grantor ID is expensive.

The second case is when the entry doesn't exist and it was created for this policy decision. In this case, if we can't find the grantor ID, we should drop this solicitation to add a policy decision and log.